File: src/optical_element/MATLAB/Furion_spherica_Mirror.m
classdef Furion_spherica_Mirror<w_oe
    %波动光学圆柱镜  
    properties
        r; %圆柱镜面曲率半径     
    end
    
    methods
        function obj=Furion_spherica_Mirror( beam_in,ds,di,chi,theta,surface,r,grating)   
            obj = obj@w_oe( beam_in,ds,di,chi,theta,surface,grating) ;  %初始化父类  optical_element的参数      
            obj.r=r;%初始化半径     
        
            obj.reflect;%反射光束          
        end
    end   
    
      methods %光学追迹
        function tracing(obj)
           obj.g_mirror = g_Furion_spherical_Mirror( obj.gbeam_in,obj.ds,obj.di,obj.chi,obj.theta,no_surfe(),obj.r,obj.grating);
        end
      end
    
end

File: src/optical_element/Furion_spherica_Mirror.cpp
#include "Furion_spherica_Mirror.h"

using namespace Furion_NS;

Furion_Spherica_Mirror::Furion_Spherica_Mirror(Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r, Grating* grating)
    : W_Oe(beam_in, ds, di, chi, theta, surface, grating)
{
    cout << "Furion_Spherica_Mirror ³õÊ¼»¯" << endl;
    this->r = r;
}

Furion_Spherica_Mirror::~Furion_Spherica_Mirror()
{

}

void Furion_Spherica_Mirror::run(Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, Grating* grating)
{
    cout << "Furion_Spherica_MirrorµÄrun" << endl;
    reflect(beam_in, ds, di, chi, theta);
}

std::string Furion_Spherica_Mirror::tracing()
{
    cout << "Furion_Spherica_Mirror" << endl;
    this->g_FS_mirror = new G_Furion_Spherical_Mirror(this->gbeam_in, this->ds, this->di, this->chi, this->theta, this->surface, this->r, this->grating);
    g_FPC_mirror->run(this->gbeam_in, this->ds, this->di, this->chi, this->theta, this->surface, this->r, this->grating);

    return ("Furion_Spherica_Mirror");
}

